app/code/core/Mage/Log/Model/Visitor.php

```diff
@@ -101,13 +101,15 @@ public function __construct(array $data = [])
     protected function _construct()
     {
         $this->_init('log/visitor');
-        $userAgent = $this->_httpHelper->getHttpUserAgent();
         $ignoreAgents = $this->_config->getNode('global/ignore_user_agents');
         if ($ignoreAgents) {
             $ignoreAgents = $ignoreAgents->asArray();
-            if (in_array($userAgent, $ignoreAgents)) {
-                $this->_skipRequestLogging = true;
-            }
+            $userAgent = $this->_httpHelper->getHttpUserAgent();
+            array_walk($ignoreAgents, function ($item) use ($userAgent) {
+                if (stristr($userAgent, $item)) {
+                    $this->_skipRequestLogging = true;
+                }
+            });
```

Comment:
```suggestion
            foreach ($ignoreAgents as $ignoreAgent) {
                if (stristr($userAgent, $ignoreAgent)) {
                    $this->_skipRequestLogging = true;
                    break;
                }
            })
```

Reply:
Or `stripos($userAgent, $ignoreAgent) !== false`?

Reply:
Just an idea ... when log is disabled, there is no need to check user-agent ...

```php
    /**
     * Object initialization
     */
    protected function _construct()
    {
        $this->_init('log/visitor');
        if ($this->_logCondition->isLogDisabled() || $this->isIgnoredUserAgent()) {
            $this->_skipRequestLogging = true;
        }
    }

    /**
     * @return bool
     */
    protected function isIgnoredUserAgent(): bool
    {
        $ignoreAgents = $this->_config->getNode('global/ignore_user_agents');
        if ($ignoreAgents) {
            $userAgent = $this->_httpHelper->getHttpUserAgent();
            foreach ($ignoreAgents->asArray() as $ignoreAgent) {
                if (stripos($userAgent, $ignoreAgent) !== false) {
                    return true;
                }
            }
        }

        return false;
    }
```

Reply:
@fballiano are you concerned about the new method?